kubernetes.go

```diff
@@ -76,18 +76,20 @@ type KubernetesClusterCreateRequest struct {
 
 	NodePools []*KubernetesNodePoolCreateRequest `json:"node_pools,omitempty"`
 
-	MaintenancePolicy *KubernetesMaintenancePolicy `json:"maintenance_policy"`
-	AutoUpgrade       bool                         `json:"auto_upgrade"`
-	SurgeUpgrade      bool                         `json:"surge_upgrade"`
+	MaintenancePolicy      *KubernetesMaintenancePolicy      `json:"maintenance_policy"`
+	AutoUpgrade            bool                              `json:"auto_upgrade"`
+	SurgeUpgrade           bool                              `json:"surge_upgrade"`
+	ControlPlanePermission *KubernetesControlPlanePermission `json:"control_plane_permission,omitempty"`
 }
 
 // KubernetesClusterUpdateRequest represents a request to update a Kubernetes cluster.
 type KubernetesClusterUpdateRequest struct {
-	Name              string                       `json:"name,omitempty"`
-	Tags              []string                     `json:"tags,omitempty"`
-	MaintenancePolicy *KubernetesMaintenancePolicy `json:"maintenance_policy,omitempty"`
-	AutoUpgrade       *bool                        `json:"auto_upgrade,omitempty"`
-	SurgeUpgrade      bool                         `json:"surge_upgrade,omitempty"`
+	Name                   string                            `json:"name,omitempty"`
+	Tags                   []string                          `json:"tags,omitempty"`
+	MaintenancePolicy      *KubernetesMaintenancePolicy      `json:"maintenance_policy,omitempty"`
+	AutoUpgrade            *bool                             `json:"auto_upgrade,omitempty"`
+	SurgeUpgrade           bool                              `json:"surge_upgrade,omitempty"`
+	ControlPlanePermission *KubernetesControlPlanePermission `json:"control_plane_permission,omitempty"`
```

Comment:
Sanity check: will `omitempty` semantics combined with the `KubernetesControlPlanePermission` type not being pointerized work for our case?

Genuinely asking because this part of the Go JSON library is always a bit scary, and mistakes can be very hard to adjust once the API is in use.

Reply:
Since `Enabled` takes a pointer and does not use `omitempty`, you should get the behavior (I'm assuming) you want.

```
Enabled          *bool    `json:"enabled"`
```

https://go.dev/play/p/nSgNNF_c1Lx